tests/cli/test_cli_core.py

```diff
@@ -173,46 +221,67 @@ def test_delete_nonexisting(cli_runner, trailblazer_context, caplog):
+    trailblazer_db.update_ongoing_analyses()
 
-        # GIVEN an analysis that is ongoing
-        analysis_obj = trailblazer_context["trailblazer"].get_latest_analysis(case_id="escapedgoat")
+    # GIVEN an analysis that is ongoing
+    analysis: Optional[Analysis] = trailblazer_db.get_latest_analysis_for_case(
+        case_name=ongoing_analysis_case_name
+    )
 
-        # WHEN trying to delete ongoing analysis without --force flag
-        result = cli_runner.invoke(delete, [str(analysis_obj.id)], obj=trailblazer_context)
+    # WHEN trying to delete ongoing analysis without --force flag
+    result = cli_runner.invoke(delete, [str(analysis.id)], obj=trailblazer_context)
 
-        # THEN command executes successfully
-        assert result.exit_code == 0
+    # THEN command executes successfully
+    assert result.exit_code == process_exit_success
 
-        # THEN error log informs user that ongoing analysis cannot be deleted without --force flag
-        assert "--force" in caplog.text
+    # THEN error log informs user that ongoing analysis cannot be deleted without --force flag
+    assert "--force" in caplog.text
 
-        # THEN analysis should not be cancelled
-        assert analysis_obj.status != "canceled"
+    # THEN analysis should not be cancelled
+    assert analysis.status != TrailblazerStatus.CANCELLED
 
 
-def test_delete_ongoing_force(cli_runner, trailblazer_context, caplog):
-    with caplog.at_level("INFO"):
-        # GIVEN Trailblazer database with analyses and jobs and an ongoing analysis
-        trailblazer_context["trailblazer"].update_ongoing_analyses()
-        analysis_obj = trailblazer_context["trailblazer"].get_latest_analysis(case_id="escapedgoat")
+def test_delete_ongoing_force(
+    cli_runner: CliRunner,
+    trailblazer_context: Dict[str, MockStore],
+    caplog,
+    ongoing_analysis_case_name,
+    process_exit_success: int,
+):
+    caplog.set_level("INFO")
 
-        # WHEN running command with --force flag
-        result = cli_runner.invoke(
-            delete, [str(analysis_obj.id), "--force"], obj=trailblazer_context
-        )
+    # GIVEN Trailblazer database with analyses and jobs and an ongoing analysis
+    trailblazer_db: MockStore = trailblazer_context["trailblazer"]
+    trailblazer_db.update_ongoing_analyses()
+    analysis: Optional[Analysis] = trailblazer_db.get_latest_analysis_for_case(
+        case_name=ongoing_analysis_case_name
+    )
 
-        # THEN command runs successfully
-        assert result.exit_code == 0
+    # WHEN running command with --force flag
+    result = cli_runner.invoke(delete, [str(analysis.id), "--force"], obj=trailblazer_context)
 
-        # THEN log informs user that Trailblazer is deleting analysis
-        assert "Deleting" in caplog.text
+    # THEN command runs successfully
+    assert result.exit_code == process_exit_success
 
+    # THEN log informs user that Trailblazer is deleting analysis
+    assert "Deleting" in caplog.text
```

Comment:
maybe an "assert analysis.status == TrailblazerStatus.CANCELLED"

Reply:
It is gone, but added this:
```
    # THEN the analysis should have been deleted from the database
    assert not trailblazer_db.get_latest_analysis_for_case(case_name=analysis.family)

```